src/communication/websocket/session.cpp

```diff
@@ -173,4 +211,21 @@ void Session::OnRead(const boost::beast::error_code ec, const size_t /*bytes_tra
   DoRead();
 }
 
+bool Session::IsAuthenticated() const noexcept { return authenticated_ || !auth_.HasAnyUsers(); }
+
+void Session::DoShutdown() {
+  std::visit(utils::Overloaded{[this](SSLWebSocket &ssl_ws) {
+                                 boost::beast::get_lowest_layer(ssl_ws).expires_after(std::chrono::seconds(30));
```

Comment:
Why `expires_after` is necessary to be called on the `tcp_stream` here, but not in case of `PlainText`?

Reply:
I based it on this example:
https://www.boost.org/doc/libs/1_77_0/libs/beast/example/advanced/server-flex/advanced_server_flex.cpp
I would say that the timeout options from `ws` don't apply to the TCP operations, so you should call it before doing anything on the layers below `Websocket`. That reminded me that we are missing the timeout for the handshake.